inference.py

```diff
@@ -1,61 +1,107 @@
+import os
 from vllm import LLM, SamplingParams
 from human_eval.data import write_jsonl, read_problems
 from pathlib import Path
 from time import time
 from human_eval import evaluation
 from os import environ
-import uuid
+import coolname
 import json
+import yaml
 
-
-def run_generation(out_file, sampling_params, llm_params):
+def get_prompts():
     problems = read_problems()
     prompts = [problem["prompt"] for problem in problems.values()]
     task_ids = list(problems.keys())
+    
+    return prompts, task_ids
 
+def run_generation(out_file, config):
+    sampling_params = config["sampling"]
+    llm_params = config["llm_params"]
+    prompts, task_ids = get_prompts()
+    
     llm = LLM(**llm_params)
     t0 = time()
     outputs = llm.generate(prompts, SamplingParams(**sampling_params))
-    print(f"generation time: {time() - t0:.3f}")
+    elapsed_time = time() - t0
+    print(f"generation time: {elapsed_time:.3f}")
 
     samples = []
     for tid, output in zip(task_ids, outputs):
         for out in output.outputs:
             samples.append(dict(task_id=tid, completion=out.text))
     write_jsonl(out_file, samples)
+    
+    return elapsed_time
 
+def create_experiment_dir(base_path="/raid/shared/llm-inference-scaling/experiments"):
+    base_path = Path(base_path)
+    experiment_name = '-'.join(coolname.generate())
+    experiment_path = base_path / experiment_name
+    experiment_path.mkdir(exist_ok=True, parents=True)
+    
+    return experiment_name, experiment_path
 
-def run_experiment(sampling_params, llm_params, evaluate=False):
-    environ["CUDA_VISIBLE_DEVICES"] = "3"  # todo do this differently
-    environ["TOKENIZERS_PARALLELISM"] = "true"
-
-    out_path = Path("/raid/shared/llm-inference-scaling/outputs")
-    out_path.mkdir(exist_ok=True, parents=True)
-
-    name = f"{uuid.uuid4()}.jsonl"  # choose out file name randomly
-    this_params = dict(sampling_params=sampling_params, llm_params=llm_params)
-    experiments_file = out_path / "_experiments.json"
+def store_metadata(name, config, base_path="/raid/shared/llm-inference-scaling/experiments"):
+    experiments_file = Path(base_path) / "experiments.json"
     if experiments_file.exists():
         with open(experiments_file, "r") as f:
             experiments = json.load(f)
     else:
         experiments = {}
-    if this_params in experiments.values():
-        return
-    experiments[name] = this_params
-    out_file = out_path / name
-    run_generation(out_file, sampling_params, llm_params)
+    experiments[name] = config
+    
+    return experiments, experiments_file
```

Comment:
Does this work if `experiments_file` doesn't exist yet? Later you call:
`with open(experiments_file, "w") as f:`
`    json.dump(experiments, f, indent=4)`
Does open work on a not yet existing file? If not the experiments_file should be created here in case it does not exist yet.

Reply:
He said he tested this and `experiments_file = Path(base_path) / "experiments.json"` creates the file if it does not exist yet. Than this is good the way it is.